certbot-apache/certbot_apache/augeasparser.py

```diff
@@ -253,6 +286,72 @@ def _aug_get_block_name(self, path):
         name = name.split("[")[0]
         return name
 
+    def _aug_resolve_child_position(self, name, position):
+        """
+        Helper function that iterates through the immediate children and figures
+        out the insertion path for a new AugeasParserNode.
+
+        Augeas also generalizes indices for directives and comments, simply by
```

Comment:
what does "generalizes indices" mean?

Reply:
I probably should have opened the context a bit, but I'll do it here.
So if we have a config file `/etc/apache2/apache2.conf` with the following contents:
```
# First comment line
# Second comment line
WhateverDirective whatevervalue
<ABlock>
    DirectiveInABlock dirvalue
</ABlock>
SomeDirective somedirectivevalue
<ABlock>
    AnotherDirectiveInABlock dirvalue
</ABlock>
# Yet another comment
```
It translates to augeas tree nodes as follows, when calling `aug.match("/files/etc/apache2/apache2.conf/*")`:

```
[
"/files/etc/apache2/apache2.conf/#comment[1]",
"/files/etc/apache2/apache2.conf/#comment[2]",
"/files/etc/apache2/apache2.conf/directive[1]",
"/files/etc/apache2/apache2.conf/ABlock[1]",
"/files/etc/apache2/apache2.conf/directive[2]",
"/files/etc/apache2/apache2.conf/ABlock[2]",
"/files/etc/apache2/apache2.conf/#comment[3]"
]
```
So regardless of directive name, its key in the Augeas tree is always "directive", same goes with the comments, and all of them have their own respective indices. Blocks however have their actual name in the Augeas path.

In this comment with generalizing I meant "hiding" the actual name of the directive or a comment.

Reply:
actually having this example as a whole has been pretty helpful for reviewing the other ones and might be useful to have in the comment

Reply:
I added the example to the main pydoc of the file. I also added a short description with an example how case sensitivity works within Augeas paths.